beacon-chain/blockchain/receive_block.go

```diff
@@ -157,29 +163,42 @@ func (s *Service) ReceiveAttesterSlashing(ctx context.Context, slashing *ethpb.A
 	s.InsertSlashingsToForkChoiceStore(ctx, []*ethpb.AttesterSlashing{slashing})
 }
 
-func (s *Service) handlePostBlockOperations(b interfaces.ReadOnlyBeaconBlock) error {
+// prunePostBlockOperationPools only runs on new head otherwise should return a nil.
+func (s *Service) prunePostBlockOperationPools(ctx context.Context, root [32]byte) error {
+	headRoot, err := s.HeadRoot(ctx)
```

Comment:
I think there will be a dead lock. `HeadRoot` requires a read lock

Reply:
I thought what potuz mentioned here https://github.com/prysmaticlabs/prysm/pull/12085#discussion_r1129209275 was to use this lock instead.

Reply:
`HeadRoot` does not require a read lock, the function takes one.

Reply:
Ah ok. `HeadRoot` uses `headLock` from Blockchain pkg not `ForkchoiceLock`. Confusing how there are two different locks